PWGHF/TableProducer/HFDsToKKPiCandidateSelector.cxx

```diff
@@ -0,0 +1,154 @@
+#include "PWGHF/DataModel/HFCandidateSelectionTables.h"
+#include "Common/Core/TrackSelectorPID.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong3;
+using namespace o2::analysis::hf_cuts_ds_tokkpi;
+
+/// Struct for applying Ds to KKpi selection cuts
+struct HfDsTokkpiCandidateSelector {
+  Produces<aod::HFSelDsToKKPiCandidate> hfSelDsToKKPiCandidate;
+
+  Configurable<double> pTCandMin{"pTCandMin", 1., "Lower bound of candidate pT"};
+  Configurable<double> pTCandMax{"pTCandMax", 36., "Upper bound of candidate pT"};
+  // TPC
+  Configurable<bool> requireTPC{"requireTPC", true, "Flag to require a positive Number of found clusters in TPC"};
+  Configurable<double> pidTPCMinpT{"pidTPCMinpT", 0.15, "Lower bound of track pT for TPC PID"};
+  Configurable<double> pidTPCMaxpT{"pidTPCMaxpT", 20., "Upper bound of track pT for TPC PID"};
+  Configurable<double> nSigmaTPC{"nSigmaTPC", 3., "Nsigma cut on TPC"};
+  // Configurable<double> TPCNClsFindablePIDCut{"TPCNClsFindablePIDCut", 50., "Lower bound of TPC findable clusters for good PID"};
+  //  TOF
+  Configurable<double> pidTOFMinpT{"pidTOFMinpT", 0.15, "Lower bound of track pT for TOF PID"};
+  Configurable<double> pidTOFMaxpT{"pidTOFMaxpT", 20., "Upper bound of track pT for TOF PID"};
+  Configurable<double> nSigmaTOF{"nSigmaTOF", 3., "Nsigma cut on TOF"};
+  // topological cuts
+  Configurable<std::vector<double>> pTBins{"pTBins", std::vector<double>{hf_cuts_ds_tokkpi::pTBins_v}, "pT bin limits"};
+  Configurable<LabeledArray<double>> cuts{"Ds_to_K_K_Pi_cuts", {hf_cuts_ds_tokkpi::cuts[0], npTBins, nCutVars, pTBinLabels, cutVarLabels}, "Ds candidate selection per pT bin"};
+
+  /// Candidate selections
+  /// \param candidate is candidate
+  /// \param trackPion1 is the first track with the pion hypothesis
+  /// \param trackKaon is the track with the kaon hypothesis
+  /// \param trackPion2 is the second track with the pion hypothesis
+  /// \return true if candidate passes all cuts
+  template <typename T1, typename T2>
+  bool selection(const T1& candidate, const T2& trackPion1, const T2& trackKaon, const T2& trackPion2)
+  {
+    auto candpT = candidate.pt();
+    int pTBin = findBin(pTBins, candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+    // check that the candidate pT is within the analysis range
+    if (candpT < pTCandMin || candpT > pTCandMax) {
+      return false;
+    }
+    // cut on daughter pT
+    if (trackPion1.pt() < cuts->get(pTBin, "pT Pi") || trackKaon.pt() < cuts->get(pTBin, "pT K") || trackPion2.pt() < cuts->get(pTBin, "pT Pi")) {
+      return false;
+    }
+    // invariant-mass cut
+    if (std::abs(InvMassDsKKPi(candidate) - RecoDecay::getMassPDG(pdg::Code::kDs)) > cuts->get(pTBin, "deltaM")) {
+      return false;
+    }
+    if (std::abs(InvMassDsPiKK(candidate) - RecoDecay::getMassPDG(pdg::Code::kDs)) > cuts->get(pTBin, "deltaM")) {
+      return false;
+    }
```

Comment:
This seems wrong. You reject the candidate if it doesn't satisfy both mass hypotheses.